nautobot/nautobot_data_validation_engine/models.py

```diff
@@ -0,0 +1,355 @@
+            raise ValidationError(
+                {
+                    "field": f"Not a valid field for content type {self.content_type.app_label}.{self.content_type.model}."
+                }
+            )
+
+        blacklisted_field_types = (
+            models.AutoField,
+            models.BigAutoField,
+            models.BooleanField,
+            models.FileField,
+            models.FilePathField,
+            models.ForeignKey,
+            models.ImageField,
+            models.JSONField,
+            models.Manager,
+            models.ManyToManyField,
+            models.NullBooleanField,
+            models.OneToOneField,
+            models.fields.related.RelatedField,
+            models.SmallAutoField,
+            models.UUIDField,
+        )
+
+        model_field = self.content_type.model_class()._meta.get_field(self.field)
+
+        if self.field.startswith("_") or not model_field.editable or isinstance(model_field, blacklisted_field_types):
+            raise ValidationError({"field": "This field's type does not support regular expression validation."})
+
+
+@extras_features(
+    "custom_fields",
+    "custom_links",
+    "export_templates",
+    "graphql",
+    "relationships",
+    "statuses",
+    "webhooks",
+)
+class MinMaxValidationRule(ValidationRule):
+    """A type of validation rule that applies min/max constraints to a given numeric model field."""
+
+    field = models.CharField(
+        max_length=50,
+    )
+    min = models.FloatField(
+        null=True, blank=True, help_text="When set, apply a minimum value contraint to the value of the model field."
+    )
+    max = models.FloatField(
+        null=True, blank=True, help_text="When set, apply a maximum value contraint to the value of the model field."
+    )
+
+    clone_fields = ["enabled", "content_type", "min", "max", "error_message"]
+
+    class Meta:
+        """Model metadata for the MinMaxValidationRule model."""
+
+        ordering = ("name",)
+        unique_together = [["content_type", "field"]]
+
+    def get_absolute_url(self, api=False):
+        """Absolute url for the instance."""
+        return reverse("nautobot_data_validation_engine:minmaxvalidationrule", args=[self.id])
+
+    def clean(self):
+        """Ensure field is valid for the model and has not been blacklisted."""
+        if self.field not in [f.name for f in self.content_type.model_class()._meta.get_fields()]:
+            raise ValidationError(
+                {
+                    "field": f"Not a valid field for content type {self.content_type.app_label}.{self.content_type.model}."
+                }
+            )
+
+        whitelisted_field_types = (
+            models.DecimalField,
+            models.FloatField,
+            models.IntegerField,
+        )
+
+        blacklisted_field_types = (
+            models.AutoField,
+            models.BigAutoField,
+        )
```

Comment:
nit: `whitelist` and `blacklist` terminology are out of fashion these days, consider changing to `allowed_` and `excluded_` or similar?

Reply:
Changed to `allowed_field_types` and `excluded_field_types`